File: app/src/main/java/net/sunniwell/app/linktaro/nettv/view/CustomResDialog.java
package net.sunniwell.app.linktaro.nettv.view;

import android.app.Dialog;
import android.content.Context;
import android.view.Window;
import android.view.WindowManager.LayoutParams;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.TextView;

public class CustomResDialog extends Dialog {
    private Button mBtnResCancel;
    private Button mBtnResOk;
    private Context mContext;
    private FrameLayout mFlResBg;
    private TextView mTvResContent;

    public CustomResDialog(Context context, int theme) {
        super(context, theme);
        this.mContext = context;
        setContentView(C0395R.layout.reservation_dialog);
        initUi();
    }

    private void initUi() {
        this.mFlResBg = (FrameLayout) findViewById(C0395R.C0397id.fl_resdialog_bg);
        this.mTvResContent = (TextView) findViewById(C0395R.C0397id.tv_res_content);
        this.mBtnResOk = (Button) findViewById(C0395R.C0397id.btn_res_ok);
        this.mBtnResCancel = (Button) findViewById(C0395R.C0397id.btn_res_cancel);
        Window dialogWindow = getWindow();
        LayoutParams lp = dialogWindow.getAttributes();
        dialogWindow.setGravity(49);
        lp.y = 200;
        dialogWindow.setAttributes(lp);
    }

    public void setDialogBg(int bgResource) {
        this.mFlResBg.setBackgroundResource(bgResource);
    }

    public void setTextContent(String text) {
        this.mTvResContent.setText(text);
    }

    public void setOkBgResource(int bgResource) {
        this.mBtnResOk.setBackgroundResource(bgResource);
    }

    public void setCancelBgResource(int bgResource) {
        this.mBtnResCancel.setBackgroundResource(bgResource);
    }

    public void setOkClickListener(OnClickListener listener) {
        this.mBtnResOk.setOnClickListener(listener);
    }

    public void setCancelClickListener(OnClickListener listener) {
        this.mBtnResCancel.setOnClickListener(listener);
    }

    public void show() {
        super.show();
        this.mBtnResOk.requestFocus();
    }
}

File: app/src/main/java/com/eliving/tv/customview/CustomResDialog.kt
package com.eliving.tv.customview

import android.app.Dialog
import android.content.Context
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import com.eliving.tv.R


class CustomResDialog(private val mContext: Context, theme: Int) :
    Dialog(mContext, theme) {
    private var mBtnResCancel: Button? = null
    private var mBtnResOk: Button? = null
    private var mFlResBg: FrameLayout? = null
    private var mTvResContent: TextView? = null
    private fun initUi() {
        mFlResBg = findViewById<View>(R.id.fl_resdialog_bg) as FrameLayout
        mTvResContent = findViewById<View>(R.id.tv_res_content) as TextView
        mBtnResOk = findViewById<View>(R.id.btn_res_ok) as Button
        mBtnResCancel =
            findViewById<View>(R.id.btn_res_cancel) as Button
        val dialogWindow = window
        val lp = dialogWindow!!.attributes
        dialogWindow.setGravity(49)
        lp.y = 200
        dialogWindow.attributes = lp
    }

    fun setDialogBg(bgResource: Int) {
        mFlResBg!!.setBackgroundResource(bgResource)
    }

    fun setTextContent(text: String?) {
        mTvResContent!!.text = text
    }

    fun setOkBgResource(bgResource: Int) {
        mBtnResOk!!.setBackgroundResource(bgResource)
    }

    fun setCancelBgResource(bgResource: Int) {
        mBtnResCancel!!.setBackgroundResource(bgResource)
    }

    fun setOkClickListener(listener: View.OnClickListener?) {
        mBtnResOk!!.setOnClickListener(listener)
    }

    fun setCancelClickListener(listener: View.OnClickListener?) {
        mBtnResCancel!!.setOnClickListener(listener)
    }

    override fun show() {
        super.show()
        mBtnResOk!!.requestFocus()
    }

    init {
        setContentView(R.layout.reservation_dialog)
        initUi()
    }
}